### PR TITLE
bloc for login event

### PR DESCRIPTION
**Pull request by lantrannk**

### CODE REVIEW

File: lib/features/auth/bloc/auth_state.dart [hunk 1]
```diff
@@ -12,9 +12,9 @@ class LoginInitial extends AuthState {}
 class LoginLoading extends AuthState {}
 
 class LoginError extends AuthState {
-  const LoginError(this.error);
+  const LoginError([this.error]);
```
Comment (nhandoanngoc-agilityio) [inline lib/features/auth/bloc/auth_state.dart]: I think we should use `LoginError({this.error})`

File: lib/features/auth/bloc/auth_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  AuthBloc() : super(LoginInitial()) {
+    on<LoginEvent>(_handleLoginEvent);
+  }
+
+  Future<void> _handleLoginEvent(
+    LoginEvent event,
+    Emitter<AuthState> emit,
+  ) async {
+    try {
+      final List<User> users = await UserStorage.getUsers();
```
Comment (nhandoanngoc-agilityio) [inline lib/features/auth/bloc/auth_bloc.dart:23]: We should emit loading before get  a user.

File: lib/features/auth/bloc/auth_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import 'package:salon_appointment/core/storage/user_storage.dart';
+
+import '../../../core/validations/validations.dart';
+import '../model/user.dart';
+
+part 'auth_event.dart';
+part 'auth_state.dart';
+
+class AuthBloc extends Bloc<AuthEvent, AuthState> {
+  AuthBloc() : super(LoginInitial()) {
```
Comment (ningo-agilityio) [inline lib/features/auth/bloc/auth_bloc.dart]: Remove LoginInitial

File: lib/features/auth/bloc/auth_bloc.dart [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  ) async {
+    try {
+      final List<User> users = await UserStorage.getUsers();
+
+      emit(LoginLoading());
+      if (FormValidation.isLoginSuccess(
+          users, event.phoneNumber, event.password)) {
+        emit(LoginSuccess());
+      } else {
+        emit(const LoginError());
```
Comment (ningo-agilityio) [inline lib/features/auth/bloc/auth_bloc.dart]: No message for error

**Review by ningo-agilityio — APPROVED** [review]

**Pull request merged**